language/move-prover/stackless-bytecode-generator/src/stackless_bytecode_generator.rs

```diff
@@ -744,71 +834,79 @@ impl<'a> StacklessBytecodeGenerator<'a> {
                 let temp_index = self.temp_count;
-                self.local_types.push(SignatureToken::Bool);
+                self.local_types.push(Type::Primitive(PrimitiveType::Bool));
                 self.temp_count += 1;
                 self.temp_stack.push(temp_index);
                 self.code.push(StacklessBytecode::Exists(
                     temp_index,
                     operand_index,
                     *struct_index,
-                    None,
+                    vec![],
                 ));
             }
 
             Bytecode::ExistsGeneric(idx) => {
                 let struct_instantiation = self.module.struct_instantiation_at(*idx);
                 let operand_index = self.temp_stack.pop().unwrap();
                 let temp_index = self.temp_count;
-                self.local_types.push(SignatureToken::Bool);
+                self.local_types.push(Type::Primitive(PrimitiveType::Bool));
                 self.temp_count += 1;
                 self.temp_stack.push(temp_index);
                 self.code.push(StacklessBytecode::Exists(
                     temp_index,
                     operand_index,
                     struct_instantiation.def,
-                    Some(struct_instantiation.type_parameters),
+                    self.get_type_params(struct_instantiation.type_parameters),
                 ));
             }
 
             Bytecode::MutBorrowGlobal(idx) | Bytecode::ImmBorrowGlobal(idx) => {
                 let struct_definition = self.module.struct_def_at(*idx);
+                let is_mut = matches!(bytecode, Bytecode::MutBorrowGlobal(..));
 
                 let operand_index = self.temp_stack.pop().unwrap();
                 let temp_index = self.temp_count;
-                self.local_types
-                    .push(SignatureToken::MutableReference(Box::new(
-                        SignatureToken::Struct(struct_definition.struct_handle),
-                    )));
+                self.local_types.push(Type::Reference(
+                    is_mut,
+                    Box::new(self.module_env.globalize_signature(&SignatureToken::Struct(
+                        struct_definition.struct_handle,
+                    ))),
+                ));
                 self.temp_stack.push(temp_index);
                 self.temp_count += 1;
                 self.code.push(StacklessBytecode::BorrowGlobal(
                     temp_index,
                     operand_index,
                     *idx,
-                    None,
+                    vec![],
                 ));
             }
 
             Bytecode::MutBorrowGlobalGeneric(idx) | Bytecode::ImmBorrowGlobalGeneric(idx) => {
                 let struct_instantiation = self.module.struct_instantiation_at(*idx);
                 let struct_definition = self.module.struct_def_at(struct_instantiation.def);
+                let is_mut = matches!(bytecode, Bytecode::MutBorrowGlobalGeneric(..));
```

Comment:
Where is matches! defined?

Reply:
std::matches. Just came along it today for the first time.